### PR TITLE
frontend/fix: #1406 added pincode in overlay sheet

### PR DESCRIPTION
**Pull request by aryanguptaSG**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Add PinCode of pickup and drop locations in overlay sheet in driver app .
figma :- [link](https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=62317%3A17717&t=5tX4hcKpIk5HTjGF-1)
output :- [1](https://drive.google.com/file/d/1d9rovC20lt-TeAcBCWkWV6efA6I0MGNw/view?usp=sharing) , [2](https://drive.google.com/file/d/1CdGSOdXaiPS_4ZOXgu52DqCNY-BnC_nc/view?usp=sharing) , [3](https://drive.google.com/file/d/1CShUdqnGo_7GVJRfxtcdWOAKPIslr0eq/view?usp=sharing) , [4](https://drive.google.com/file/d/16EVFLr8vxhHuDSdY27mZm1fGmEZhldKD/view?usp=sharing)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Comment by kavya-shree-s** [comment]
@aryanguptaSG Can you attach screenshot/screenrecording

**Comment by Rohit4997** [comment]
> @aryanguptaSG Can you attach screenshot/screenrecording

Also record specialZone View as well in this case

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/NotificationUtils.java [hunk 1]
```diff
@@ -231,6 +231,8 @@ public static void showAllocationNotification (Context context, String title, St
                     sheetData.putInt("rideRequestPopupDelayDuration",entity_payload.has("rideRequestPopupDelayDuration")?entity_payload.getInt("rideRequestPopupDelayDuration"):0);
                     sheetData.putInt("customerExtraFee",(entity_payload.has("customerExtraFee") && !entity_payload.isNull("customerExtraFee") ? entity_payload.getInt("customerExtraFee") : 0));
                     sheetData.putInt("keepHiddenForSeconds", (entity_payload.has("keepHiddenForSeconds") && !entity_payload.isNull("keepHiddenForSeconds") ? entity_payload.getInt("keepHiddenForSeconds") : 0));
+                    sheetData.putString("sourcePinCode", addressPickUp.getString("areaCode"));
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/NotificationUtils.java:234]: If `addressPickUp` is not having any areaCode, Handle that case as well

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/OverlaySheetService.java [hunk 2]
```diff
@@ -144,6 +144,18 @@ public void onViewHolderBind(SheetAdapter.SheetViewHolder holder, int position,
             holder.destinationAddress.setText(model.getDestinationAddress());
             holder.textIncPrice.setText(String.valueOf(model.getNegotiationUnit()));
             holder.textDecPrice.setText(String.valueOf(model.getNegotiationUnit()));
+            if(model.getSourcePinCode() != null &&  model.getSourcePinCode().trim().length()>0){
+                holder.sourcePinCode.setText(model.getSourcePinCode().trim());
+                holder.sourcePinCode.setVisibility(View.VISIBLE);
+            }else{
+                holder.sourceAddress.setMaxLines(2);
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/OverlaySheetService.java:151]: `holder.sourcePinCode.setVisibility(View.GONE);`
Can you add this line in else case

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/OverlaySheetService.java [hunk 3]
```diff
@@ -144,6 +144,18 @@ public void onViewHolderBind(SheetAdapter.SheetViewHolder holder, int position,
...
+            }
+            if(model.getDestinationPinCode() != null && model.getDestinationPinCode().trim().length()>0){
+                holder.destinationPinCode.setText(model.getDestinationPinCode());
+                holder.destinationPinCode.setVisibility(View.VISIBLE);
+            }else{
+                holder.destinationAddress.setMaxLines(2);
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/OverlaySheetService.java:157]: `holder.destinationPinCode.setVisibility(View.GONE);`
Add this line in else case

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestActivity.java [hunk 4]
```diff
@@ -258,7 +258,9 @@ public void run() {
                         rideRequestBundle.getInt("rideRequestPopupDelayDuration"),
                         negotiationUnit,
                         rideRequestBundle.getInt("customerTip"),
-                        rideRequestBundle.getString("specialLocationTag"));
+                        rideRequestBundle.getString("specialLocationTag"),
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestActivity.java:261]: Add the pincode visibility conditions in this Activity's `onBindViewHolder` as well

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by aryan-13 — APPROVED** [review]

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestActivity.java (hunk 4 above)
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestActivity.java:261]: Please add the pincode in this Activity as well

**Review by witcher-shailesh — APPROVED** [review]

**Review by kavya-shree-s — APPROVED** [review]

**Review by Rohit4997 — APPROVED** [review]

**Pull request merged**